internal/cmd/egctl/config_test.go

```diff
@@ -267,3 +274,257 @@ func sampleAggregatedConfigDump(configDump protoreflect.ProtoMessage) aggregated
 		},
 	}
 }
+
+type fakeCLIClient struct {
+	pods []corev1.Pod
+	cm   *corev1.ConfigMap
+}
+
+func (f *fakeCLIClient) RESTConfig() *rest.Config {
+	return nil
+}
+
+func (f *fakeCLIClient) Pod(types.NamespacedName) (*corev1.Pod, error) {
+	return nil, nil
+}
+
+func (f *fakeCLIClient) PodsForSelector(string, ...string) (*corev1.PodList, error) {
+	return &corev1.PodList{Items: f.pods}, nil
+}
+
+func (f *fakeCLIClient) PodExec(types.NamespacedName, string, string) (stdout string, stderr string, err error) {
+	return "", "", nil
+}
+
+func (f *fakeCLIClient) Kube() kubernetes.Interface {
+	return fake.NewSimpleClientset(f.cm)
+}
+
+func TestFetchRunningRateLimitPods(t *testing.T) {
+
+	cases := []struct {
+		caseName      string
+		rlPods        []corev1.Pod
+		namespace     string
+		labelSelector []string
+		expectErr     error
+	}{
+		{
+			caseName: "normally obtain the rate limit pod of Running phase",
```

Comment:
the log shows there's something wrong with this test case: https://github.com/envoyproxy/gateway/actions/runs/8047676612/job/21977431309?pr=2674#step:4:144.

seems the conditions in this status also needs to be updated according to `checkRateLimitPodStatusReady`